client/client.go

```diff
@@ -81,6 +82,30 @@ func New(ctx context.Context, address string, opts ...ClientOpt) (*Client, error
 		if sd, ok := o.(*withSessionDialer); ok {
 			sessionDialer = sd.dialer
 		}
+		if opt, ok := o.(*withOutgoingMeta); ok {
+			customDialOptions = append(customDialOptions,
```

Comment:
We should append these to `unary` and `stream` from above I think, just for consistency.